packages/fast-form-generator-react/src/form/form-item.array.tsx

```diff
@@ -98,10 +121,38 @@ class FormItemArray extends React.Component<IFormItemArrayProps & IManagedClasse
         );
     }
 
+    public componentDidUpdate(): void {
+        this.state.hideOptionMenu
+            ? window.removeEventListener("click", this.handleWindowClick)
+            : window.addEventListener("click", this.handleWindowClick);
+    }
+
+    private storeMenuRef = (ref?: HTMLUListElement): void => {
+        if (ref) {
+            this.menuElement = ref;
+        }
+    }
+
+    private storeButtonRef = (ref?: HTMLButtonElement): void => {
```

Comment:
similarly to the renaming I suggested above, I would have this be called setOptionMenuTriggerRef, store isn't really standard.